### ISSUE TITLE
[supernova] sound input fails if number of requested output channels exceeds soundcard capabilities

### ISSUE THREAD

**Issue by dyfer** [supercollider/supercollider#4029]
Environment
-----------

* Your SuperCollider version: 3.10dev
* Your operating system and version: macOS 10.13

Steps to reproduce (for bugs)
-----------------------------

```supercollider
//set your default devices to internal soundcard (2 inputs/outputs)
Server.supernova;
s.options.numOutputBusChannels_(4); //choose number larger than the soundcard provides
s.boot;
s.meter; //no sound from the input
//moreover, the input busses are unusable
x = {PinkNoise.ar(-48.dbamp)}.play(outbus: s.options.numOutputBusChannels); //signal accumulates

x.free;
s.quit;

```

Error message (for bugs)
------------------------

no error

Expected Behavior
-----------------

Sound input should work, even if specified number of output channels exceeds number of channels available in the hardware.

Current Behavior
----------------

No signal from audio interface appears on the input busses.
If one attempts playing a signal on an input bus from a synth, the signal weirdly accumulates and eventually distorts.

Actually, it sounds almost exactly like an impulse at control rate:
```SuperCollider
x = {Impulse.ar(ControlRate.ir) * -12.dbamp}.play;
```
You can listen to the input to check (careful, this will quickly overload/distort!)
```SuperCollider
y = {SoundIn.ar(0)}.play;
x = {PinkNoise.ar(-48.dbamp)}.play(outbus: s.options.numOutputBusChannels); //signal accumulates
s.freqscope;

x.free; y.free;
```

Note: this applies to supernova on macOS, which uses PortAudio. I also tested with scsynth compiled with PortAudio (on macOS) but was not able to reproduce the issue, so it's not a purely PortAudio issue, but supernova / PortAudio.

I have not tested on other platforms (I have a feeling it might affects supernova on Windows, but that doesn't currently build so...).

**Comment by sonoro1234** [comment]
This can surely be repaired on the PR for separate input and output in portaudio for supernova?

**Comment by dyfer** [comment]
@sonoro1234 do you have an idea what the reason for this bug is? I was poking around here but couldn't find anything obvious.

**Comment by sonoro1234** [comment]
I am not having this bug in Lua2SC so could be sclang side?

**Comment by dyfer** [comment]
My intuition is that there's something wrong with how samples are copied to the PA buffer. Number of channels, once it gets to exchanging data between supernova and PA, is clipped to the hardware's number of channels. That itself is the same as scsynth, so the problem is not likely be with that in itself. But again, I don't have a full understanding of this...

**Comment by brianlheim** [comment]
fixed in #4575, thanks @dyfer !
